app/lib/features/space/pages/pins_page.dart

```diff
@@ -19,81 +19,106 @@ class SpacePinsPage extends ConsumerWidget {
 
   @override
   Widget build(BuildContext context, WidgetRef ref) {
+    final spaceRelationsOverview =
+        ref.watch(spaceRelationsOverviewProvider(spaceIdOrAlias));
     final space = ref.watch(spaceProvider(spaceIdOrAlias)).requireValue;
     final pins = ref.watch(spacePinsProvider(space));
-    // get platform of context.
-    return DecoratedBox(
-      decoration: const BoxDecoration(gradient: AppTheme.primaryGradient),
-      child: CustomScrollView(
-        slivers: [
-          SliverToBoxAdapter(
-            child: SpaceHeader(
-              spaceIdOrAlias: spaceIdOrAlias,
-            ),
-          ),
-          SliverToBoxAdapter(
-            child: Padding(
-              padding: const EdgeInsets.all(8.0),
-              child: Row(
-                children: [
-                  Expanded(
-                    child: Text(
-                      'Pins',
-                      style: Theme.of(context).textTheme.titleMedium,
-                    ),
+
+    return spaceRelationsOverview.when(
+      data: (spaceData) {
+        bool checkPermission(String permission) {
+          if (spaceData.membership != null) {
+            return spaceData.membership!.canString(permission);
+          }
+          return false;
+        }
+
+        final canPostPin = checkPermission('CanPostPin');
+
+        // get platform of context.
+        return DecoratedBox(
+          decoration: const BoxDecoration(gradient: AppTheme.primaryGradient),
+          child: CustomScrollView(
+            slivers: [
+              SliverToBoxAdapter(
+                child: SpaceHeader(
+                  spaceIdOrAlias: spaceIdOrAlias,
+                ),
+              ),
+              SliverToBoxAdapter(
+                child: Padding(
+                  padding: const EdgeInsets.all(8.0),
+                  child: Row(
+                    children: [
+                      Expanded(
+                        child: Text(
+                          'Pins',
+                          style: Theme.of(context).textTheme.titleMedium,
+                        ),
+                      ),
+                      Visibility(
+                        visible: canPostPin,
+                        child: IconButton(
+                          icon: Icon(
+                            Atlas.plus_circle_thin,
+                            color: Theme.of(context).colorScheme.neutral5,
+                          ),
+                          iconSize: 28,
+                          color: Theme.of(context).colorScheme.surface,
+                          onPressed: () => context.pushNamed(
+                            Routes.actionAddPin.name,
+                            queryParameters: {'spaceId': spaceIdOrAlias},
+                          ),
+                        ),
+                      ),
```

Comment:
If we do that entire spaceRelationsOverview only to get the canPostPin figured out here, we probably do not want to wait for that to show the pins. Instead I'd suggest to replace the Visibility with a Consumer or a widget specific that then does the provider call ..

looking at that and how often we have that pattern we might want to create a common reusable Widget `UserCan(roomId, canString, child: IconButton( ... ) )` that does a `roomProviderMembership`-lookup, checks for that `canString` and if that comes with false, renders a `SizedBox.shrink()`.

Reply:
Good notice.
I will create reusable components for this.